ssl/quic/quic_channel.c

```diff
@@ -1331,8 +1331,19 @@ static int ch_on_transport_params(const unsigned char *params,
     ossl_unused uint64_t rx_max_idle_timeout = 0;
     ossl_unused const void *stateless_reset_token_p = NULL;
     QUIC_PREFERRED_ADDR pfa;
+    SSL_CONNECTION *sc = SSL_CONNECTION_FROM_SSL(ch->tls);
 
-    if (ch->got_remote_transport_params) {
+    /*
+     * When HRR happens the client sends the transport params in the new client
+     * hello again. Reset the transport params here and load them again.
+     */
+    if (sc->hello_retry_request != SSL_HRR_NONE && ch->got_remote_transport_params) {
+        ch->max_local_streams_bidi = 0;
+        ch->max_local_streams_uni = 0;
+        ch->got_local_transport_params = 0;
+        OPENSSL_free(ch->local_transport_params);
+        ch->local_transport_params = NULL;
```

Comment:
Should we in this case actually check that the transport params as sent in the new client hello are the same as the original ones? Hmm... I did not find anything in regards to that in RFCs so perhaps this is OK as is.

Reply:
I'm not sure, as the RFC does not mentions this case (at least I did not found it)

Reply:
The CI is failing on the "test setup" when it sets the group list. Which I don't understand why at the moment..

When I try to just return 1 and don't reset the option, the failure remains the same tho